lstm.go

```diff
@@ -1,40 +1,132 @@
 package golgi
 
 import (
+	"fmt"
+
 	"github.com/chewxy/hm"
-	G "gorgonia.org/gorgonia"
+	"gorgonia.org/gorgonia"
 	"gorgonia.org/tensor"
 )
 
-// LSTM represents a static LSTM model
-type LSTM struct{}
+func newLSTM(g *gorgonia.ExprGraph, layer *LSTMLayer, name string) (lp *LSTM) {
+	var l LSTM
+	l.g = g
+	l.name = name
+	l.input = layer.getInput(g, name)
+	l.forget = layer.getInput(g, name)
+	l.output = layer.getInput(g, name)
+	l.cell = layer.getInput(g, name)
+	return &l
+}
+
+// LSTM represents an LSTM RNN
+type LSTM struct {
+	name string
+
+	g *gorgonia.ExprGraph
+
+	input  whb
+	forget whb
+	output whb
+	cell   whb
+}
+
+func (l *LSTM) getGate(w *whb, inputVector, prevHidden *gorgonia.Node, act ActivationFunction) (gate *gorgonia.Node, err error) {
```

Comment:
I see what is happening here but I think `getGate` is not a good name. What you're doing is forwarding through a gate. `fwdGate` or `activateGate` might be better. The latter feels more august - like you're some sort of gatekeeper activating a portal.

Reply:
![](https://i.imgur.com/A6iZyk8.gif)

Reply:
activateGate sounds awesome